engine/execution/provider/engine_test.go

```diff
@@ -188,14 +168,8 @@ func TestProviderEngine_onChunkDataRequest(t *testing.T) {
 			}).
 			Return(nil)
 
-		execState.
-			On("GetBlockIDByChunkID", chunkID).
-			Return(blockID, nil)
-		execState.
-			On("ChunkDataPackByChunkID", mock.Anything, chunkID).
-			Return(chunkDataPack, nil)
-
-		execState.On("GetCollection", chunkDataPack.CollectionID).Return(&collection, nil)
```

Comment:
Retrieving and composing a `ChunkDataPack` happens in the `ExecutionState`, and the created `ChunkDataPack` also contains a `Collection`. So no more collection retrieval is needed.